File: core/identity-service/src/main/java/com/gov/crypto/identityservice/service/AuthService.java
package com.gov.crypto.identityservice.service;

import com.gov.crypto.model.User;
import com.gov.crypto.repository.UserRepository;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.stereotype.Service;

@Service
public class AuthService {

    private final UserRepository repository;
    private final PasswordEncoder passwordEncoder;

    // JWT removed - using Redis session-based auth

    public AuthService(UserRepository repository, PasswordEncoder passwordEncoder) {
        this.repository = repository;
        this.passwordEncoder = passwordEncoder;
    }

    public String saveUser(User user) {
        user.setPasswordHash(passwordEncoder.encode(user.getPasswordHash()));
        repository.save(user);
        return "User added to system";
    }

    // generateToken removed - sessions handle authentication
}

File: backend/identity-service/src/main/kotlin/com/gov/crypto/identityservice/service/AuthService.kt
package com.gov.crypto.identityservice.service

import com.gov.crypto.identityservice.entity.User
import com.gov.crypto.identityservice.repository.UserRepository
import org.springframework.security.crypto.password.PasswordEncoder
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
class AuthService(
    private val userRepository: UserRepository,
    private val passwordEncoder: PasswordEncoder
) {

    @Transactional
    fun register(username: String, passwordRaw: String): User {
        if (userRepository.findByUsername(username) != null) {
            throw IllegalArgumentException("Username already exists")
        }
        
        val hash = passwordEncoder.encode(passwordRaw)
        val user = User(
            username = username, 
            passwordHash = hash!!,
            checkSum = "SHA3-256-placeholder"
        )
        return userRepository.save(user)
    }

    fun authenticate(username: String, passwordRaw: String): Boolean {
        val user = userRepository.findByUsername(username) ?: return false
        return passwordEncoder.matches(passwordRaw, user.passwordHash)
    }
}
